### PR TITLE
docs(readme): added description new structure and onboarding new product teams and special cases

### PR DESCRIPTION
**Pull request by FaGru3n**
Added documentation after refactoring new structure of onboarding repository

### CODE REVIEW

File: PREREQUISITES.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Our current environments are running in Azure Cloud based on Azure Kubernetes Service and were managed or deployed with Terraform scripts. The following Tools are required to run theese scripts.
+
+## Azure CLI
+
+Using our Azure Subscription you have to be invited to our Subscription and installed Azure CLI.
+Links to our Subsrciption ()
+
+- [Azure CLI Installing Guidelines](https://learn.microsoft.com/en-us/cli/azure)
+
+You had to be invited to our Azure Subscription on [Azure-Portal](https://portal.azure.com/#@catenax.onmicrosoft.com)
```
Comment (almadigabor) [inline PREREQUISITES.md:12]:
```suggestion
You have to be invited to our Azure Subscription on [Azure-Portal](https://portal.azure.com/#@catenax.onmicrosoft.com)
```

File: PREREQUISITES.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Login to Azure on CLI for help, just ask an System Team Internal Member who has allready done that.
```
Comment (almadigabor) [inline PREREQUISITES.md:14]:
```suggestion
Login to Azure on CLI for help, just ask a System Team Internal Member who has already done that.
```

File: PREREQUISITES.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Reload your Terminal / CLI  based on your Working OS
+# Testing if autocompletion is working install / init should appeare
+terraform i
+# Switch to a terraform file-directory for e.g our github-folder and initialize terraform
+terraform init
+```
+
+- enable your CLI
+
+Depending on which terraform run you will about to plan and perform you need to export the desired `TOKEN` from the terraform run to generate the ressources within it.
```
Comment (almadigabor) [inline PREREQUISITES.md:41]: I'd remove this for easier understanding if the next sentence is modified.

```suggestion
```

File: PREREQUISITES.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+For `terraform apply` and `terraform plan` command the following command line variables has to be set:
```
Comment (almadigabor) [inline PREREQUISITES.md:43]:
```suggestion
For `terraform apply` and `terraform plan` command the following command line variables has to be exported from our `Vault` and set:
```

File: README.md [hunk 5]
```diff
@@ -1,40 +1,47 @@
...
-- Hashicorp Vault
-  - Product related secret engine
-  - Authentication methods (AppRole)
+This repository contains all necessary parts to onboard a Catena-X product to consortia environments.
 
-## GitHub
+## Kubernetes / Argocd related Parts
 
-### Folder Structure
+In `cluster` folder are all available clusters, in the `cluster/<clustername>/kustomization.yaml` we describe which product is deployed on the each cluster with the decleared permission level for the product on the target cluster.
```
Comment (almadigabor) [inline README.md:7]:
```suggestion
The `cluster` folder contains all available clusters, in the `cluster/<clustername>/kustomization.yaml` we describe on which cluster the products with the declared permissions are deployed.
```

File: README.md [hunk 6]
```diff
@@ -1,40 +1,47 @@
...
+> and it is declared on the cluster  
+> `cluster/devsecops-testing/kustomization.yaml`
+
+More Information can be found under argocd Folder: [README.md](argocd/README.md)
+
+## GitHub related parts
+
+We are providing all github repositorys and github organisation teams via an `terraform.tfvars`.
+
+Discription  of the to be added github teams / github repositorys / github repository-teams-mapping can be found in `github/terraform.tfvars` and if we want to create a new Repo/Team/Mapping we need to add all of the following.
```
Comment (almadigabor) [inline README.md:27]:
```suggestion
Description  of the to be added GitHub teams / GitHub repositories / GitHub repository-teams-mapping can be found in `github/terraform.tfvars` and if we want to create a new Repo/Team/Mapping we need to add all of the following.
```

File: README.md [hunk 7]
```diff
@@ -1,40 +1,47 @@
...
 
-### Description
+- 'New' Github Repository
+- 'New' Github Teams
```
Comment (almadigabor) [inline README.md:30]:
```suggestion
- 'New' Github Team
```

File: README.md [hunk 8]
```diff
@@ -1,40 +1,47 @@
...
+- 'New' Mapping between Github Repository and Github Team
 
-tbd
+More Information can be found under Github Folder: [README.md](github/README.md)
```
Comment (almadigabor) [inline README.md:33]:
```suggestion
More information can be found under the github folder: [README.md](github/README.md)
```

File: README.md [hunk 9]
```diff
@@ -1,40 +1,47 @@
...
 
-> note:  
-> Design and implementation phase!
+## Vault related parts
 
-## Kubernetes
+We are providing for all product-teams vaults with their related Key/Value pairs based on the gh-product-team-name.
```
Comment (almadigabor) [inline README.md:37]:
```suggestion
We are providing all product-teams `Vault` access with their Secrets Engine where they can store their own  Key/Value pairs based on the `gh-product-team-name`.
```

File: README.md [hunk 10]
```diff
@@ -1,40 +1,47 @@
...
+
+Discription  of the to be added product vault can be found in `vault/terraform.tfvars`.
```
Comment (almadigabor) [inline README.md:39]:
```suggestion
Description of the product to be added to vault can be found in `vault/terraform.tfvars`.
```

File: README.md [hunk 11]
```diff
@@ -1,40 +1,47 @@
...
+
+More Information can be found under Vault folder: [README](vault/README.md)
```
Comment (almadigabor) [inline README.md:41]:
```suggestion
More information can be found under vault folder: [README](vault/README.md)
```

File: README.md [hunk 12]
```diff
@@ -60,32 +67,46 @@ tbd
...
+- `argocd/product-example/base-read-only/resources/avp-secret.yaml`
+- `cluster/devsecops-testing/kustomization.yaml`
 
-tbd
+create a PR for your branch. After the PR has been approved and merged into main branch, the new team will be
+automatically deployed to cluster environment (via ArgoCD application over the `_<ENVIRONMENT>-config_` at ArgoCD within the `_CORE_` cluster).
 
-> note:  
-> Design and implementation phase!
+Only after the merge in Github and the terraform apply have been done, the terraform state is consistent.
```
Comment (almadigabor) [inline README.md:111]:
```suggestion
Only after the merge in GitHub and the `terraform apply` have been done, the terraform state is consistent, otherwise changes which are applied in parallel by someone else might be deleted again.
```

File: README.md [hunk 13]
```diff
@@ -60,32 +67,46 @@ tbd
...
+Otherwise changes which are applied in parallel by someone else might be deleted again.
```
Comment (almadigabor) [inline README.md:112]:
```suggestion
```

File: argocd/README.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
+## ArgoCD
+
+To provide a product-team access to the and cluster e.g. devsecops-testing infrastructure following onboarding steps must be performed (all
```
Comment (almadigabor) [inline argocd/README.md:3]:
```suggestion
To provide a product-team access to a cluster (e.g. devsecops-testing) the following onboarding steps must be performed (all
```

File: argocd/README.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+steps are related to repository [product-onboarding](https://github.com/catenax-ng/product-onboarding/blob/main/README.md)):
+
+with this we provide a
```
Comment (almadigabor) [inline argocd/README.md:6]:
```suggestion
With this we provide a 
```

File: argocd/README.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+- new ArgoCD project for new product-team
+- create a new AVP secret for new product-team
+- new k8s namespace for new product-team
+
+First create a new branch based on branching naming convention in [product-onboarding](https://github.com/catenax-ng/product-onboarding) repo for onboarding a new
+product-team to ArgoCD.
+
+### Create ArgoCD Project
+
+Create terraform manifests for the new product-team into a separate folder in this `argocd` folder and this will create per default a:
```
Comment (almadigabor) [inline argocd/README.md:17]:
```suggestion
Create terraform manifests for the new product-team into a separate folder inside the `argocd` folder and this will create per default a:
```

File: argocd/README.md [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+  VAULT_ADDR: https://vault.demo.catena-x.net/
+  AVP_TYPE: vault
+  AVP_AUTH_TYPE: approle
+  AVP_ROLE_ID: <role_id>
+  AVP_SECRET_ID: <secret_id>
+
+```
+The secret will be called _vault-secret_ and stored in k8s namespace related to product-team.
+
+Store the manifests in [product-onboarding](https://github.com/catenax-ng/product-onboarding)repo in path `argocd` and an entry based on their permissions in every `cluster/<ClusterEnvironment>/kustomization.yaml`  where tha new product-team need a new argocd-project. Default is dev and int (named here Hotel-Budapest).
```
Comment (almadigabor) [inline argocd/README.md:81]:
```suggestion
Store the manifests in [product-onboarding](https://github.com/catenax-ng/product-onboarding) repo in path `argocd` and an entry based on their permissions in every `cluster/<ClusterEnvironment>/kustomization.yaml`  where the new product-team need a new argocd-project. Default is dev and int (named here Hotel-Budapest).
```

File: argocd/README.md [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+
+### Prepare Deployment Of ArgoCD Project And AVP Secret
+
+To deploy k8s namespace, ArgoCD Project and the AVP secret to on cluster envionment e.g devsecops-testing  you'll have to add a new entry into `cluster/devsecops-testing/kustomization.yaml`
```
Comment (almadigabor) [inline argocd/README.md:85]:
```suggestion
To deploy k8s namespace, ArgoCD Project and the AVP secret to the cluster (e.g devsecops-testing) you'll have to add a new entry to the `cluster/devsecops-testing/kustomization.yaml` file.
```

**Review by almadigabor — CHANGES_REQUESTED** [review]
Read through the first 3 files and added some suggestions with syntactic and semantic fixes. Will check the other files tomorrow.

**Comment by FaGru3n** [comment]
> Read through the first 3 files and added some suggestions with syntactic and semantic fixes. Will check the other files tomorrow.

Thx @almadigabor updated the most typo's. -  "four eyes see more than two" 😉

File: cluster/README.md [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+- [beta](/cluster/beta)
+- [dev](/cluster/dev)
+- [devsecops](/cluster/devsecops-testing/)
+- [int](/cluster/int)
+- [pre-prod](/cluster/pre-prod)
+
+## Include all products on this cluster and its permission level
+
+In `cluster` folder are all available clusters, in the `cluster/<clustername>/kustomization.yaml` we describe which product is deployed on the each cluster with the decleared permission level for the product on the target cluster.
```
Comment (almadigabor) [inline cluster/README.md:15]: Just copied this from the main README.md

```suggestion
The `cluster` folder contains all available clusters, in the `cluster/<clustername>/kustomization.yaml` we describe on which cluster the products with the declared permissions are deployed.
```

File: github/README.md [hunk 20]
```diff
@@ -1,5 +1,161 @@
-This folder contains Terraform scripting for GitHub onboarding.
+# This folder contains Terraform scripting for GitHub onboarding
 
-Possibilities:
-- Single subfolder for each product (one terraform run per product)
-- flat, without subfolders as we're used to (one terraform run for all)
\ No newline at end of file
+**TLDR** one terraform run for all gh-repositorys gh-teams and gh-mapping between repositorys and teams
```
Comment (almadigabor) [inline github/README.md:3]:
```suggestion
**TLDR** one terraform run for all gh-repositories, gh-teams and gh-mapping between repositories and teams
```

File: github/README.md [hunk 21]
```diff
@@ -1,5 +1,161 @@
...
+
+In our case we have one Github Organisation which is capable for all github repository information within the catena-x ng for each product.
```
Comment (almadigabor) [inline github/README.md:5]:
```suggestion
In our case we have one GitHub Organization which contains all GitHub repository information within
 catena-x ng for each product.
```

File: github/README.md [hunk 22]
```diff
@@ -1,5 +1,161 @@
...
+## Assigning a team as contributor to a repository via terraform
+
+Contribution access to a repository in our GitHub organization is granted on a team level. We do not
+grant this kind of access to individuals.
+Access is again managed by our terraform variables under [github folder](..github/).
+
+To manage contribution access for a team on a repository, edit the `terraform.tfvars` file in the `github` directory.
+
+There, add a new map entry to the `github_repositories_teams` variable inside `terraform.tfvars`.
+As convention, we decided to for the map key as a combination of repository and team (`<repository-name-team-name>`).
```
Comment (almadigabor) [inline github/README.md:16]:
```suggestion
As convention, we decided for the map key to be a combination of repository and team (`<repository-name-team-name>`).
```

File: github/README.md [hunk 23]
```diff
@@ -1,5 +1,161 @@
...
+This is done, because we have cases of multiple teams contributing to a single repository. This is configured, by
+adding multiple entries to the `github_repositories_teams` map, containing the same repository, but a different team
+each time.
+
+As default, we configure `maintain` access on the product repositories for the teams, since all the administrative
+tasks are handled by the team managing the organization.
+
+## GitHub Organistation
```
Comment (almadigabor) [inline github/README.md:24]:
```suggestion
## GitHub Organization
```

File: github/README.md [hunk 24]
```diff
@@ -1,5 +1,161 @@
...
+
+The following section describes how to handle users, teams and repositories in our GitHub organisation
```
Comment (almadigabor) [inline github/README.md:26]:
```suggestion
The following section describes how to handle users, teams and repositories in our GitHub organization
```

File: github/terraform.tfvars [hunk 25]
```diff
@@ -1,13 +1,17 @@
-#github_teams = {
-#    "argocdadmins" : {
-#      "name" : "argocdadmins",
-#      "description" : "ArgoCD OAuth administrator team"
+# Explanation for the Structure of the Variables with provide for terraform scripts
```
Comment (almadigabor) [inline github/terraform.tfvars:1]:
```suggestion
# Explanation for the structure and variables we provide for terraform scripts
```

File: vault/README.md [hunk 26]
```diff
@@ -1,5 +1,26 @@
-This folder contains Terraform scripting for Vault onboarding.
+# This folder contains Terraform scripting for Vault onboarding
 
-Possibilities:
-- Single subfolder for each product (one terraform run per product)
-- flat, without subfolders as we're used to (one terraform run for all)
\ No newline at end of file
+**TLDR** one terraform run for all vault permissions for each product
+
+In our case we have one Vault Cluster which is capable for all Vault information for each cluster and product.
```
Comment (almadigabor) [inline vault/README.md:5]:
```suggestion
In our case we have one Vault Cluster which contains all secrets for each cluster and product.
```

File: vault/terraform.tfvars [hunk 27]
```diff
@@ -1,3 +1,17 @@
+# New Product Object for Vault for authenticate new product gh teams to the ArgoCD deployed Vault 
+# on UI and a technical user for new product
```
Comment (almadigabor) [inline vault/terraform.tfvars:2]: Move one line up

```suggestion
```

File: vault/terraform.tfvars [hunk 28]
```diff
@@ -1,3 +1,17 @@
+# New Product Object for Vault for authenticate new product gh teams to the ArgoCD deployed Vault
```
Comment (almadigabor) [inline vault/terraform.tfvars:1]:
```suggestion
# New product object for product teams to authenticate in Vault. Also contains the name of the machine user, the name of the secret engine and the Kubernetes secret with which the ArgoCD Vault Plugin can authenticate to resolve secret variables.
```

**Review by almadigabor — CHANGES_REQUESTED** [review]
Some small changes for the remaining files.

**Review by almadigabor — APPROVED** [review]
Looks good.

**Review by SebastianBezold — APPROVED** [review]
